transport/tcp/serve_tcp_ipmap.go

```diff
@@ -0,0 +1,113 @@
+package tcp
+
+import (
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/zrepl/zrepl/transport"
+)
+
+type ipMapEntry struct {
+	ip     net.IP
+	ident  string
+	subnet *net.IPNet
+	zone   string
+}
+
+type ipMap struct {
+	entries []ipMapEntry
+}
+
+func ipMapFromConfig(clients map[string]string) (*ipMap, error) {
+	entries := make([]ipMapEntry, 0, len(clients))
+	for clientInput, clientIdent := range clients {
+		clientIP, clientCIDR, zone := handleIPV6(clientInput)
+
+		if clientCIDR != nil {
+			if len(clientIdent) == 1 && clientIdent == "*" {
+				clientIdent = "willbeclientip"
+			}
+			if len(clientIdent) > 1 && clientIdent[len(clientIdent)-1:] == "*" {
+				clientIdent = strings.TrimRight(clientIdent, "*")
```

Comment:
Please add a test case that tests a mapping a la `"10.1.2.0/24":"my-*-server"`, which should evaluate like this: `"10.1.2.3":"my-10.1.2.3-server"`.

Also, there should  be a test cases for:

```
"10.1.2.3":"too-lazy-to-repeat-left-hand-side-*" => 10.1.2.3 maps to "too-lazy-to-repeat-left-hand-side-10.1.2.3*

"10.1.2.0/24":"expression-without-star" => expectInitErr = true
```